src/Microsoft.Azure.WebJobs.Host/FunctionTimeoutException.cs

```diff
@@ -33,17 +34,23 @@ protected FunctionTimeoutException(SerializationInfo info, StreamingContext cont
         }
 
         /// <inheritdoc/>
-        public FunctionTimeoutException(string message, Guid instanceId, string methodName, TimeSpan timeout, Exception innerException)
+        public FunctionTimeoutException(string message, Guid instanceId, string methodName, TimeSpan timeout, Task functionTask, Exception innerException)
             : base(message, instanceId, methodName, innerException)
         {
             Timeout = timeout;
+            FunctionTask = functionTask;
         }
 
         /// <summary>
         /// The function timeout value that expired.
         /// </summary>
         public TimeSpan Timeout { get; set; }
 
+        /// <summary>
+        /// The task that did not complete due to a timeout.
+        /// </summary>
+        public Task FunctionTask { get; set; }
```

Comment:
nit: I wonder if you shouldn't just name this "Task", like you use "Timeout" above, rather than "FunctionTimeout"